### ISSUE TITLE
Support for scaling custom resources and statefulset

### ISSUE THREAD

**Issue by raulsh** [kedacore/http-add-on#438]
Based on the docs and issues, there is no way to scale another kind of resources besides Deployments.  We need scale rollouts (CRD of Argo Rollouts). Even, we use core KEDA to scale rollouts.

### Use-Case

We use Argo Rollouts, so in Keda we can specify the Kind of the resource (https://keda.sh/docs/2.6/concepts/scaling-deployments/). This addon must scale any CRD/Resource with /scale subresource (Statefulset or Argo Rollout resource).

I use rollouts to deploy some HTTP workloads.

### Specification

Probably same as core KEDA, with parameters to specify the kind and the apiVersion of CRDs

**Comment by tomkerkhove** [comment]
Just out of curiosity - How is HTTP scaling required for Argo Rollouts? (I'm not an argo expert)

**Comment by raulsh** [comment]
Rollouts resources are similar to native Deployment resources. They are managed by Argo Rollout controller. The difference is that rollouts support Blue/Green and Canary deployments (and other functions). This addon must scale any CRD/Resource with /scale subresource (Statefulset or Argo Rollout resource).

**Comment by tomkerkhove** [comment]
And those rollouts are being consumed over HTTP then? I'm just asking because I'm not sure why this is a problem for the HTTP add-on and why vanilla KEDA can't help you on this as that's supported there.

**Comment by raulsh** [comment]
Yes... we use to deploy HTTP workloads (APIs, monoliths, etc). But, why the question? That's not the addon purpose? or am I misunderstanding something? (i'm not a english native speaker, sorry)

In fact, this is the perfect solution to our case. Because we have HTTP workloads that have multiple containers, but doesnt have a centralized queue to manage request and manage pending requests properly (while scaling workload). Because we have a high dynamic request throughput in our internal services. And I think this is the case that this addon want to cover.

Of course we can use vanilla Keda, but the "Interceptor" design is the solution that fit to our needs. And the main issue for us Is because we are using rollouts instead native deployments haha.

**Comment by tomkerkhove** [comment]
> Yes... we use to deploy HTTP workloads (APIs, monoliths, etc). But, why the question? That's not the addon purpose? or am I misunderstanding something? (i'm not a english native speaker, sorry)

I was just curious about your scenario because my lack of experience with Argo Rollouts :)

> In fact, this is the perfect solution to our case. Because we have HTTP workloads, but doesnt have a centralized queue to manage request and manage pending requests properly (while scaling workload). Because we have a high dynamic request throughput in our internal services. And I think this is the case that this addon want to cover.
> 
> Of course we can use vanilla Keda, but the "Interceptor" design is the solution that fit to our needs. And the main issue for us Is because we are using rollouts instead native deployments haha.

Then you are in the right place! Let's add this to our features and see when we can add this - Thanks!

**Comment by miguelvr** [comment]
@tomkerkhove I would like to work on this. From my pass on the code, it's a fairly big change that probably requires a design proposal. What is the process to propose a design for a new feature?

**Comment by tomkerkhove** [comment]
Sounds like a good idea! We typically use Google Docs so that everyone can add comments ([example](https://docs.google.com/document/d/14Z04AWiMBkSX88uiaVS-kg-8QjjNcDhVy6-u5FMUk_w/edit))

**Comment by prakarsh-dt** [comment]
I am a maintainer at [Devtron](https://github.com/devtron-labs/devtron) and was exploring some kubernetes native tool that could help scale workloads from 0 on getting a HTTP request. I stumbled upon this tool and it looks promising and we have already done our POCs. 

I would like to integrate KEDA HTTP Add-on ( HTTPScaledObject ) into Devtron, we would indeed need support for custom resources, particularly Argo Rollout. We are eager to contribute to this project by developing the necessary capabilities and sharing them back with the community.

Since there hasn't been any recent activity on this feature request thread for over 6 months, I would appreciate it if you could provide an update on the current status. It would be helpful if someone could guide me to any existing work that has been done in this direction. I am keen to collaborate and ensure that our integration aligns with the ongoing efforts of the project.

**Comment by tomkerkhove** [comment]
There was no progress on this one so I think we need a design proposal first before we can get started on this one

**Comment by JorTurFer** [comment]
Hi @prakarsh-dt 
Are you still interested on this? I can take a look and maybe we can integrate them (I don't promise anything), of course, if you are willing to do it from your side and open a PR, we'll be happy

**Comment by prakarsh-dt** [comment]
> Hi @prakarsh-dt Are you still interested on this? I can take a look and maybe we can integrate them (I don't promise anything), of course, if you are willing to do it from your side and open a PR, we'll be happy

Nope @JorTurFer I haven't made any progress on this, you can pick it.
